tests/test_dataserver.py

```diff
@@ -1,8 +1,32 @@
+from pathlib import Path
 from hifz.dataserver import DataServer
 from hifz.models import Card
 
+def test_dataserver_returns_card_csv():
+    server = DataServer()
+    temp_file_path = "data/test_arabic.csv"
```

Comment:
Use a tmp_path fixture 

[docs](https://docs.pytest.org/en/stable/how-to/tmp_path.html)